CMakeGraphVizOptions.cmake

```diff
@@ -0,0 +1,4 @@
+set(GRAPHVIZ_CUSTOM_TARGETS TRUE)
```

Comment:
Is this file required? I think we have enough files in the project root directory. Maybe move it into cmake folder.

Reply:
@Chocobo1 

It's used for configuring the images produced by the `--graphviz` flag, such as https://github.com/qbittorrent/qBittorrent/pull/12746#issuecomment-623537587.

According to the CMake docs (https://cmake.org/cmake/help/latest/module/CMakeGraphVizOptions.html),

> This file is first searched in CMAKE_BINARY_DIR, and then in CMAKE_SOURCE_DIR. If found, the variables set in it are used to adjust options for the generated Graphviz files.

I would also prefer if CMake searched for it in the cmake modules location natively, but oh well.
So just leaving it in `CMAKE_SOURCE_DIR` is the easiest way, vs. placing the file in the cmake modules folder and copying it to `CMAKE_BINARY_DIR` at configure-time (not sure how easy it is to do and don't really want to do it, at least now), or something like this: https://marc.info/?l=kde-commits&m=149150989601551&w=2.

I too worry about the excessive number of files at the root of the project.
These are the ones that can be removed in the near-future, and are a much bigger problem than a single additional CMake-related file:

- Replaced by a `.clang-format` file:
    `.editorconfig`, `uncrustify.cfg`, `codingStyleQtCreator.xml` (contributors may configure their text editor/IDE how they like, then run clang-format before commiting; I doubt any other tools is better/more consistent than clang-format, since it is backed by the infrastructure of one of the major C++ compilers)
- Replaced by a GH Actions workflow (maybe; I'll give this a try soon-ish), which goes in inside the `.github` directory:
    `.travis.yml`, `.appveyor.yml`
- Simply removed:
    `TODO`, `INSTALL` (basic instructions to build can be in the README + a link to the wiki for the rest)
- Simply removed due to dropping `qmake + autotools` as a build system:
    `bootstrap.sh`, `conf.pri.in`, `conf.pri.windows`, `configure`, `configure.ac`, `macxconf.pri`, `qbittorrent.pro`, `unixconf.pri`, `version.pri`, `winconf.pri`

That's a reduction of 14 or 16 files, depending on whether we actually move to GH Actions for CI, or 10 by just dropping `qmake + autotools` alone.

Here's my proposal for compromise, and a start to truly solving the "bloated root" problem: https://github.com/qbittorrent/qBittorrent/pull/13208/

Reply:
>It's used for configuring the images produced by the --graphviz flag, such as #12746 (comment). The default settings are pretty bad for our use case.

It looks like a not-so-useful feature to me and I would still prefer it to be dropped.
@glassez what do you think?

Reply:
At least it should be dropped from this already bloated PR.

Reply:
@glassez @Chocobo1 

Well, I guess there is no reason why this cannot be a separate PR, so I'll remove it from this one.

Reply:
Done (forgot to actually do it before).